### ISSUE TITLE
ParserCLI in interactive mode should use Settings defaults for its SQL transformation flags

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#14176]
Most `ParserCLI` flags are set to the `Settings` default value, but this isn't the case for SQL transformation flags:

```
> /transform-patterns
Transform patterns                 : null
```

The effect is the same (as `null` settings always behave like the default), but still, this is a bit weird for users.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0. Will backport later.

**Comment by lukaseder** [comment]
Backported to 3.17.6 (https://github.com/jOOQ/jOOQ/issues/14177), 3.16.12 (https://github.com/jOOQ/jOOQ/issues/14178)
